### PR TITLE
VACMS-18418: Update search app links to va-link component

### PR DESCRIPTION
**Pull request by chriskim2311**
## Summary
This PR changes the links in the search app to use va-link.

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18418

## Testing done
Tested locally HTML markup and GA events

## Screenshots

Search_Results___Veterans_Affairs

## What areas of the site does it impact?

Search

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### CODE REVIEW

File: src/applications/search/containers/SearchApp.jsx [hunk 1]
```diff
@@ -793,63 +793,25 @@ class SearchApp extends React.Component {
...
-                  }
-                >
-                  VA handbooks and other publications
-                </a>
+                  text="VA handbooks and other publications"
+                />
               </li>
               <li>
-                <a
-                  className="right-nav-link"
```
Comment (randimays) [inline src/applications/search/containers/SearchApp.jsx:841]: Do we use this class name elsewhere? Curious if the styles can be removed.

**Review by randimays — DISMISSED** [review]
Just one comment; otherwise approved!

File: src/applications/search/containers/SearchApp.jsx (hunk 1 above)
Comment (chriskim2311) [reply src/applications/search/containers/SearchApp.jsx:841]: Nice catch I will remove it

**Review by randimays — APPROVED** [review]
LGTM

**Pull request merged**